consensus/core/src/commit_observer.rs

```diff
@@ -123,8 +133,19 @@ impl CommitObserver {
         for commit in unsent_commits {
             // Commit index must be continuous.
             assert_eq!(commit.index(), last_sent_commit_index + 1);
-            let committed_sub_dag = load_committed_subdag_from_store(self.store.as_ref(), commit);
-
+            let mut committed_sub_dag =
+                load_committed_subdag_from_store(self.store.as_ref(), commit);
+
+            // On recovery leader schedule will be updated with the current scores
+            // and the scores will be passed along with each of the commits sent to
+            // sui so that the current scores are available for submission.
+            committed_sub_dag.update_scores(
+                self.leader_schedule
```

Comment:
The scores should be computed only once outside of the for loop.

Reply:
Updated so we are only doing it once for the last subdag